lib/framedivider.rb

```diff
@@ -1,5 +1,81 @@
 require "openstudio"
 
+
+##
+# Flatten OpenStudio 3D points vs Z-axis (Z=0).
+#
+# @param [Array] pts Point3D array
+#
+# @return [Array] Flattened OS 3D points array
+def flatZ(pts)
+  raise "Invalid pts (flatZ)" unless pts
+
+  cl = OpenStudio::Point3dVector
+  valid = pts.is_a?(cl) || pts.is_a?(Array)
+  raise "#{pts.class}? expected #{cl} (flatZ)" unless valid
+
+  cl = OpenStudio::Point3d
+  pts.each do |pt|
+    raise "#{pt.class}? expected OS 3D points (flatZ)" unless pt.class == cl
+    pt = OpenStudio::Point3d.new(pt.x, pt.y, 0)
+  end
+  pts
+end
+
+##
+# Validate whether an OpenStudio subsurface can 'fit in' (vs parent, siblings)
+#
+# @param [OpenStudio::Model::Model] model An OS model
+# @param [String] id SubSurface identifier
+# @param [Array] pts Point3D array of subsurface 'id'
+#
+# @return [Bool] Returns true if subsurface can fit in.
+def fit?(model, id, pts)
+  raise "Invalid model (fit?)" unless model
+  raise "Invalid ID (fit?)" unless id
+  raise "Invalid pts (fit?)" unless pts
+
+  cl = OpenStudio::Model::Model
+  raise "#{model.class}? expected #{cl} (fit?)" unless model.is_a?(cl)
+
+  cl = OpenStudio::Point3dVector
+  valid = pts.is_a?(cl) || pts.is_a?(Array)
+  raise "#{pts.class}? expected #{cl} (flatZ)" unless valid
+
+  cl = OpenStudio::Point3d
+  pts.each do |pt|
+    raise "#{pt.class}? expected OS 3D points (fit?)" unless pt.class == cl
+  end
+
+  s = model.getSubSurfaceByName(id)
+  raise "#{id} subSurface missmatch (fit?)" if s.empty?
+  s = s.get
+
+  raise "Invalid parent surface (fit?)" if s.surface.empty?
+  parent = model.getSurfaceByName(s.surface.get.nameString)
+  raise "#{id} missing parent surface (fit?)" if parent.empty?
+  parent = parent.get
+  ft = OpenStudio::Transformation::alignFace(parent.vertices).inverse
+  ft_parent = (ft * parent.vertices).reverse
+  ft_parent = flatZ(ft_parent)
+
+  siblings = []
+  model.getSubSurfaces.each do |sub|
+    next if sub.nameString == id
+    next if sub.surface.empty?
+    next unless sub.surface.get.nameString == parent.nameString
+    siblings << flatZ( (ft * sub.vertices).reverse )
+  end
+
+  ft_pts = flatZ(ft * pts)
+  ft_pts.each do |ft_pt|
+    return false unless OpenStudio::pointInPolygon(ft_pt, ft_parent, TOL)
+    siblings.each do |sibling|
+      return false if OpenStudio::pointInPolygon(ft_pt, sibling, TOL)
+    end
+  end
```

Comment:
Also checking if new _rough opening_ vertices are overlapping any of the other subsurfaces (i.e., siblings).